hs_core/models.py

```diff
@@ -3348,9 +3356,12 @@ def size(self):
 
         Raises SessionException if iRODS fails.
         """
+        # trigger file size read for files that haven't been set yet
+        for f in self.files.filter(_size__lt=0):
+            f.size
```

Comment:
@sblack-usu I think ```f.calculate_size()``` is better in this case since ```f.size``` will check whether size is less than 0 again.

Reply:
That is a slight optimization.